sql-cli/sql_cli/dag_runner.py

```diff
@@ -80,6 +119,9 @@ def run_dag(
         session=session,
     )
     log.debug("Getting dagrun for dag %s", dag.dag_id)
+
+    dag = _get_tailored_dag(dag, task_id, include_upstream)
```

Comment:
I think the code would be easier to read if we had the following conditional here:
```
if task_id and include_upstream:
```
and only invoked the function to change the DAG if necessary.

Reply:
Unfortunately, adding more conditionals is increasing the code complexity; hence, I have moved that conditional to another method.